piece_of_cake.py

```diff
@@ -0,0 +1,16 @@
+# get the prices of the ingredients ,the amount need and the optional - which is the ingredients we don't need
+# and calculate the price.
+def get_recipe_price(prices, optionals=[], **ingredients):
+    total_price = 0
+    for ingredient in ingredients:
```

Comment:
try to think of a way to use itertools.reduce to shorten the length of this code